src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -360,19 +360,32 @@ public static bool IsEntirelyWithinCrefSyntax(this SyntaxTree syntaxTree, int po
 
         private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int position, char lastChar, CancellationToken cancellationToken)
         {
-            if (token.Kind() is not (
+            var kind = token.Kind();
+            if (kind is not (
                     SyntaxKind.StringLiteralToken or
                     SyntaxKind.CharacterLiteralToken or
                     SyntaxKind.SingleLineRawStringLiteralToken or
-                    SyntaxKind.MultiLineRawStringLiteralToken))
+                    SyntaxKind.MultiLineRawStringLiteralToken or
+                    SyntaxKind.Utf8StringLiteralToken or
+                    SyntaxKind.Utf8SingleLineRawStringLiteralToken or
+                    SyntaxKind.Utf8MultiLineRawStringLiteralToken))
             {
                 throw new ArgumentException(CSharpCompilerExtensionsResources.Expected_string_or_char_literal, nameof(token));
             }
 
+            // There wouldn't be a UTF8 string literal token if it's incomplete.
```

Comment:
should mention "because the literal would have had to end with `"u8` in the first place.